test/test_tables.py

```diff
@@ -778,6 +906,154 @@ class T5(qv.Table):
+        x = qv.Int64Column(nullable=True)
+        y = qv.Int64Column(nullable=True)
+
+    class WrapperWithNulls(qv.Table):
+        id = qv.StringColumn(nullable=True)
+        pair = PairWithNulls.as_column(nullable=True)
+
+    class DoublyNestedWithNulls(qv.Table):
+        id = qv.StringColumn()
+        inner = WrapperWithNulls.as_column(nullable=True)
+
+    # Null great-grandchild
+    dn = DoublyNestedWithNulls.from_kwargs(
+        id=["a", "b", "c"],
+        inner=WrapperWithNulls.from_kwargs(id=["a", "b", "c"], pair=PairWithNulls.from_kwargs(y=[4, 5, 6])),
+    )
+    assert pc.all(pc.equal(dn.inner.pair.x, dn.column("inner.pair.x")))
+    assert pc.all(pc.equal(dn.inner.pair.y, dn.column("inner.pair.y")))
+    assert pc.all(pc.equal(dn.inner.id, dn.column("inner.id")))
+    assert pc.all(pc.equal(dn.id, dn.column("id")))
+
+    # Null grandchild
+    dn = DoublyNestedWithNulls.from_kwargs(
+        id=["a", "b", "c"], inner=WrapperWithNulls.from_kwargs(id=["a", "b", "c"])
+    )
+    assert pc.all(pc.equal(dn.inner.pair.x, dn.column("inner.pair.x")))
+    assert pc.all(pc.equal(dn.inner.pair.y, dn.column("inner.pair.y")))
+    assert pc.all(pc.equal(dn.inner.id, dn.column("inner.id")))
+    assert pc.all(pc.equal(dn.id, dn.column("id")))
+
+    # Null child
+    dn = DoublyNestedWithNulls.from_kwargs(id=["a", "b", "c"])
+    assert pc.all(pc.equal(dn.inner.pair.x, dn.column("inner.pair.x")))
+    assert pc.all(pc.equal(dn.inner.pair.y, dn.column("inner.pair.y")))
+    assert pc.all(pc.equal(dn.inner.id, dn.column("inner.id")))
+    assert pc.all(pc.equal(dn.id, dn.column("id")))
+
+
+def test_column_empty():
+    t = Pair.empty()
+    assert len(t.column("x")) == 0
+    assert len(t.column("y")) == 0
+
+
+def test_column_nested_empty():
+    w = Wrapper.empty()
+    assert len(w.column("pair.x")) == 0
+    assert len(w.column("pair.y")) == 0
+
+
+def test_column_nested_doubly_empty():
+    class DoublyNested(qv.Table):
+        inner = Wrapper.as_column()
+
+    dn = DoublyNested.empty()
+    assert len(dn.column("inner.pair.x")) == 0
+    assert len(dn.column("inner.pair.y")) == 0
+
+
+def test_column_invalid_name():
+    t = Pair.from_kwargs(x=[1, 2, 3], y=[4, 5, 6])
+    with pytest.raises(KeyError):
+        t.column("z")
+
+
+def test_column_nested_invalid_name():
+    w = Wrapper.from_kwargs(id=["a", "b", "c"], pair=Pair.from_kwargs(x=[1, 2, 3], y=[4, 5, 6]))
+    with pytest.raises(KeyError):
+        w.column("pair.z")
+    with pytest.raises(AttributeError):
+        w.column("wrong.x")
+
+
+def test_column_nested_doubly_invalid_name():
+    class DoublyNested(qv.Table):
+        inner = Wrapper.as_column()
+
+    dn = DoublyNested.from_kwargs(
+        inner=Wrapper.from_kwargs(id=["a", "b", "c"], pair=Pair.from_kwargs(x=[1, 2, 3], y=[4, 5, 6]))
+    )
+    with pytest.raises(KeyError):
+        dn.column("inner.pair.z")
+    with pytest.raises(AttributeError):
+        dn.column("inner.wrong.x")
```

Comment:
> Please add tests, at least for `column`, of the following cases:
> 
> 3. The dot-delimited syntax specifies a column that does not exist (wrapper.column("unknown.whatever"))
> 
Tests added here though please let me know if you want consistent error types or if this behavior is okay.

Reply:
It's a little odd, but not terrible. I think this is fine.